assets/wizards/advertising/components/onboarding/index.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { useEffect, useState, useRef, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import { Card, ButtonCard, Notice, TextControl } from '../../../../components/src';
+import GoogleOAuth from '../../../connections/views/main/google';
+
+export default function AdsOnboarding( { onUpdate, onSuccess } ) {
+	const credentialsInputFile = useRef( null );
+	const [ fileError, setFileError ] = useState( '' );
+	const [ inFlight, setInFlight ] = useState( false );
+	const [ networkCode, setNetworkCode ] = useState( '' );
+	const [ isConnected, setIsConnected ] = useState( false );
+	const [ useOAuth, setUseOAuth ] = useState( null );
+
+	const updateGAMCredentials = credentials => {
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack/v1/wizard/advertising/credentials',
+			method: 'post',
+			data: { credentials, onboarding: true },
+		} )
+			.then( () => {
+				setIsConnected( true );
+				if ( typeof onSuccess === 'function' ) onSuccess();
+			} )
+			.finally( () => {
+				setInFlight( false );
+			} );
+	};
+
+	const handleCredentialsFile = event => {
```

Comment:
They should be available in both places since the onboarding setup can be exited, either by closing the modal or the browser.

Reply:
Absolutely, I did not mean that this functionality should not be available here. What I wanted to say is that it's code duplication, and it could use a refactor – most of the `handleCredentialsFile` function is generic handling of file reading. But it's not blocking.